compiler/common-artifacts/CMakeLists.txt

```
@@ -32,6 +32,19 @@ else()
     return()
   endif()
 
+  # NOTE assume only use 3.8.? or 3.10.? or 3.12.?
+  if((Python_VERSION VERSION_GREATER_EQUAL 3.8) AND (Python_VERSION VERSION_LESS 3.8.99))
+    set(PYTHON_VERSION_MINOR 8)
+  elseif((Python_VERSION VERSION_GREATER_EQUAL 3.10) AND (Python_VERSION VERSION_LESS 3.10.99))
+    set(PYTHON_VERSION_MINOR 10)
+  elseif((Python_VERSION VERSION_GREATER_EQUAL 3.12) AND (Python_VERSION VERSION_LESS 3.12.99))
+    set(PYTHON_VERSION_MINOR 12)
+  else()
```

How about using this condition?
Please note that Python 3.9 and 3.11 are not supported with this logic.
And, it seems `x` is more compatible to express the version number.


How about using this condition?
Please note that Python 3.9 and 3.11 are not supported by this logic.
Also, using `x` to represent the patch version (e.g., 3.8.x) seems more consistent and clear.

```suggestion
  # NOTE assume only use 3.8.x or 3.10.x or 3.12.x
  if((Python_VERSION VERSION_GREATER_EQUAL 3.8) AND (Python_VERSION VERSION_LESS 3.9))
    set(PYTHON_VERSION_MINOR 8)
  elseif((Python_VERSION VERSION_GREATER_EQUAL 3.10) AND (Python_VERSION VERSION_LESS 3.11))
    set(PYTHON_VERSION_MINOR 10)
  elseif((Python_VERSION VERSION_GREATER_EQUAL 3.12) AND (Python_VERSION VERSION_LESS 3.13))
    set(PYTHON_VERSION_MINOR 12)
  else()
```

---

I believe  Python 3.9 and 3.11 are not default versions on any Ubuntu LTS distro, so I believe it was @seanshpark intention, to allow only LTS default python version :)